### ISSUE TITLE
[subplat][emails] Add Email Templates and Sender Function for Subscription Renewal Reminder

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#8223]
We need to add email templates and a function to send the subscription renewal reminder email. See FXA-3166 for the copy.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-3351)
┆Attachments: <a href="https://jira.mozilla.com/secure/attachment/16410/Screen+Shot+2021-05-21+at+4.20.23+PM.png">Screen Shot 2021-05-21 at 4.20.23 PM.png</a>
┆Issue Number: FXA-3351

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

[~byee@mozilla.com] we need copy for this email.  Are you the person for this?

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

[~jecook@mozilla.com] , [~crooney@mozilla.com] : There is more context in the story ticket that this ticket fulfills,  FXA-3166, (we broke the work down into smaller tasks later on); let us know if you'd still like more information.

Actually, looking at that ticket, it looks like [~byee@mozilla.com] provided us with copy ( https://docs.google.com/document/d/1PbxlKSVt41kpU63axX6D8w0L6gnUyn1kmYzAkz0zf1U/edit?usp=sharing ) that has been approved by legal (see this comment in the story ticket).

I'll remove the `needs-content` label. Thanks Jess for following up.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Now that I'm actually in the weeds creating this new email, a couple of product/content questions arose in relation to the email copy ( https://docs.google.com/document/d/1PbxlKSVt41kpU63axX6D8w0L6gnUyn1kmYzAkz0zf1U/edit?usp=sharing ) that Brian provided:

  1. [~byee@mozilla.com]: Currently, no other transactional emails include a link to a product-specific SUMO page, and we don't have a Stripe metadata property ( https://mozilla.github.io/ecosystem-platform/docs/fxa-engineering/fxa-sub-platform#stripe-productplans ) to define that URL (i.e. something like `product:supportURL`). Can I file a follow-up issue for this and remove the "help center" link in this new email in the interim?
2. [~crooney@mozilla.com] (cc [~jecook@mozilla.com]): The email copy provided didn't have a title in the body of the email like the rest of our transactional emails (see this Miro board ( https://miro.com/app/board/o9J_lWaWaZA=/ ) for a partial set of those emails), so I added one: "Your subscription will be renewed soon" (see attached screenshot). Does that seem all right with you?

I'm going to add the `needs-product` and `needs-content` labels back to this ticket, and I will also ping you all in Slack. Unless I hear otherwise in the next couple of business days, I plan to move forward with these changes.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

QA testing STR
 (once the ticket is closed)

Note: This feature depends on FXA-3420 being closed (i.e. the `subscription-reminders` script needs to be running in Stage).

  1. With a Firefox account, sign up for a weekly plan with either Stripe or PayPal. E.g. https://accounts.stage.mozaws.net/subscriptions/products/prod_H63sY3qFAwW4Rk?plan=plan_H647TGZpWyAx4j ( https://accounts.stage.mozaws.net/subscriptions/products/prod_H63sY3qFAwW4Rk?plan=plan_H647TGZpWyAx4j ) 
  2. Wait about 24 hours.
  3. Receive an email titled "${productName} automatic renewal notice", where `${productName}` is replaced with the name of the product (e.g. Firefox Variety Product).
  4. Verify that the quantitative information for the email is correct (the reminder notice period is 6 days on Stage), the links work, and that the rest of the content looks correct (email content is specified here ( https://docs.google.com/document/d/1PbxlKSVt41kpU63axX6D8w0L6gnUyn1kmYzAkz0zf1U/edit?usp=sharing )).
  5. In the Stripe Stage dashboard ( https://dashboard.stripe.com/test/dashboard ), look up the subscription (e.g. the URL is of the form https://dashboard.stripe.com/test/subscriptions/sub_H6N4GyIYacxC35 ( https://dashboard.stripe.com/test/subscriptions/sub_H6N4GyIYacxC35 )), and verify that, under "Subscription details", the "Current period" end date is 6 days out from the date you received the email.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

I filed FXA-3597 to cover adding a product-specific support URL to Stripe metadata, so we can include a "help center" link in the future.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this on stage (Train 207) while using Firefox and Windows 10.
We observed the following:

  * no "Dear <customer_name>" section (might be because a title was added, but, we should also keep the introduction part);
  * "here" and  "contact us" redirect to manage subscription and contact support pages;

The "Current period" is correct.
I made the verification using a credit card, but I'll also check this using the PayPal payment method. Same check for a real product

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Other details we noticed:

  * all renewal emails are sent at 4:00 AM
  * we verified different products and "Current periods" are correct for all of them;

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Thanks Valentina for verifying this; sorry for the delay in replying to your questions – I was on PTO, and then it was a U.S. holiday yesterday.

> no "Dear <customer_name>" section (might be because a title was added, but, we should also keep the introduction part);

Good catch. I must have missed that when I copied the email over. I will open a PR to add "Dear ${productName} customer," in the right place and reference that PR here.

It doesn't seem like we are consistent in our email copy in terms of using a more formal style (Dear so-and-so/Sincerely so-and-so) and not (e.g. the subscriptionPaymentProviderCancelled email which was the last email we added in FXA-3165), but the asymmetry of this current email is even less desirable.

In my opinion this wouldn't warrant a dot release, in part because we are not omitting critical information, but mostly because these emails won't be going to customers in production until mid to late October.

> "here" and "contact us" redirect to manage subscription and contact support pages - correct?

That is correct.
